File: chap1/And16tst.py
# -*- coding:utf-8 -*-

code = ["load {name}.hdl, output-file {name}.out, output-list a%B1.16.1 b%B1.16.1 out%B1.16.1;\n".format(name="And16")]

for a in range(2 ** 16):
    for b in range(2 ** 16):
        code.append("set a %B" + format(a, "016b") + ", set b %B" + format(b, "016b") + ", eval, output;\n")
        print("\r"+str((b + a * 2**16) / 2**32), end="")
print()

with open("And16.tst", "w") as f:
    f.writelines(code)

File: chap1/And16tst.cpp
// Generate And16.tst

#include<bitset>
#include<cmath>
#include<iostream>
#include<fstream>

int main() {
    std::ofstream test("And16.tst");
    if(test.fail())
    {
        std::cerr << "failed to open file : And16.tst" << std::endl;
        return 0;
    }

    test << "load And16.hdl, output-file And16.out, output-list a%B1.16.1 b%B1.16.1 out%B1.16.1;" << std::endl;

    int size = std::pow(2, 16);

    for(int a=0; a < size; a++)
        for(int b=0; b < size; b++)
            test << "set a %B" << std::bitset<16>(a) << ", set b %B" << std::bitset<16>(b) << ", eval, output;" << std::endl;

  return 0;
}
